compiler/src/dotty/tools/dotc/ast/TreeInfo.scala

```diff
@@ -630,7 +630,7 @@ trait TypedTreeInfo extends TreeInfo[Type] { self: Trees.Instance[Type] =>
   /** An extractor for def of a closure contained the block of the closure. */
   object closureDef {
     def unapply(tree: Tree)(implicit ctx: Context): Option[DefDef] = tree match {
-      case Block((meth @ DefDef(nme.ANON_FUN, _, _, _, _)) :: Nil, closure: Closure) =>
+      case Block((meth : DefDef) :: Nil, closure: Closure) if meth.symbol == closure.meth.symbol =>
         Some(meth)
       case Block(Nil, expr) =>
         unapply(expr)
```

Comment:
@nicolasstucki : I changed this extractor, as some anonymous functions do not use the name `nme.ANON_FUN`.

Reply:
That looks good